tests/test_app/genesis_tb/top.svp

```diff
@@ -71,6 +94,20 @@ module top;
         .axil_ifc(axil_ifc)
     );
 
+//; if ($using_matrix_unit) {
```

Comment:
Hm I am assuming that this `for` loop is why you decided to use `genesis`. Did you consider maybe using a `generate` instead, and leaving genesis out of the Makefile? Something like

```
if (`USING_MATRIX_UNIT != 0) begin 

    //============================================================================//
    // Matrix unit outputs 
    //============================================================================//
    wire [`MU_DATAWIDTH-1:0] mu2cgra [`OC_0-1:0];
    genvar i

    generate
        for (i=0; i<`OC_0; i++) assign mu2cgra[i] = 16'b1;
    endgenerate
end
```
Also see <https://stackoverflow.com/questions/56777662/using-assign-inside-a-for-loop>

Reply:
Actually if you go down this path, you probably want the Makefile to define the macro variables this way instead
```
# Indicates if matrix unit ifc should be included; "undefined" is the default
# USING_MATRIX_UNIT = True

# Width of data coming from matrix unit
MU_DATAWIDTH ?= 32

# Matrix unit output channel unrolling
OC_0 ?= 32
```

...and then instead of `if (USING_MATRIX_UNIT != 0) begin...end`
in the "generate" example above,
you'd use something like ``` `ifdef USING_MATRIX_UNIT...`endif  ```

Reply:
(oops i guess MU_DATAWIDTH is supposed to default 16 instead of 32...)